scripts/showcase/samplesheet-prep-test.py

```diff
@@ -1,55 +1,399 @@
+
+    # Get the list of samples in the sample sheet
+    sample_list = sample_sheet_df['Sample_ID'].unique().tolist()
+
+    # A bit going on here:
+    # 1. Slim down the tracking data frame to just the Sample ID and the type of sample.
+    # 2. Rename the columns to 'Sample_ID and Sample_Type (the latter is used in the TSO500 sample sheet)
+    # 3. Filter to rows only present in this sample sheet
+    # 4. Add the column 'Type' that we then group by. TSO_DNA and TSO_RNA are set to TSO
+    slimmed_tracking_df = tracking_sheet_df[['Type', 'Sample_ID (SampleSheet)']].\
+        rename(columns={"Sample_ID (SampleSheet)": "Sample_ID",
+                        "Type": "Sample_Type"}).\
+        query("Sample_ID in @sample_list")
+
+    slimmed_tracking_df['Type'] = slimmed_tracking_df['Sample_Type'].apply(lambda x: 'TSO' if x.startswith("TSO") else x)
+
+    # Merge and return the data frames with the 'Type included
+    return pd.merge(sample_sheet_df, slimmed_tracking_df, on='Sample_ID', how='left')
+
+
+def write_sample_sheet(sample_sheet_df, sample_sheet_header_rows, sample_sheet_output_file):
+    """
+    Given a sample type, a full sample sheet df, and a header row, write out the sample,sheet as a csv file
+    """
+
+    with open(sample_sheet_output_file, 'w') as sample_sheet_output_h:
+        # Write out the header rows
+        sample_sheet_output_h.writelines(sample_sheet_header_rows)
+        # Write out [Data]
+        sample_sheet_output_h.write("{}\n".format(HEADER_LINE_PRECURSOR))
+        # Write out sample sheet
+        sample_sheet_df.\
+            to_csv(sample_sheet_output_h, sep=",", header=True, index=False)
+
+    # Completed writing of header file
+
+
+def modify_sample_sheet(sample_sheet_df, sample_sheet_header_rows, sample_type):
+    """
+    Massive if else function based on the second parameter
+    """
+
+    # Get mod strategy from sample_type
+    modification_strategy = [mod_strat for mod_strat, sample_types in VALID_SAMPLE_TYPES.items()
+                             if sample_type in sample_types]
+
+    if not len(modification_strategy) == 1:
+        logger.warning("Warning we couldn't figure out what to do with "
+                       "this dataset of type '{}', so we're leaving it as is".format(sample_type))
+        modification_strategy = None
+    else:
+        modification_strategy = modification_strategy[0]
+
+    if modification_strategy == "TSO500":
+        # Add two extra columns to sample-sheet
+        modified_sample_sheet_df = convert_sample_sheet_to_tso500(sample_sheet_df)
+        # No changes to header rows required.
+        modify_sample_header_rows = sample_sheet_header_rows
+    elif modification_strategy == "V2":
+        # Add two extra columns to sample-sheet
+        modified_sample_sheet_df = convert_sample_sheet_to_v2(sample_sheet_df)
+        # No changes to header rows required.
+        modify_sample_header_rows = convert_sample_sheet_header_to_v2(sample_sheet_header_rows)
```

Comment:
needs update to comments.

Reply:
https://github.com/umccr/infrastructure/pull/16/files#diff-9803ecbbdf15c05af6e26ccc75f88c84R342-R345

Reply:
Done, I've changed this around a little so now modification 'strategies' aren't necessarily mutually exclusive. **modification_strategy** isn't a great variable name in this case, happy to change it to something that's more descriptive of the different keys in the VALID_SAMPLE_TYPES global variable.